src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -148,7 +151,7 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
                     Debug.Assert(sslHostName == null);
                     Debug.Assert(proxyUri != null);
 
-                    _http2Enabled = false;
+                    _http2Enabled = false; // ToDo: why?
```

Comment:
We don't support HTTP/2 proxies, only HTTP/1. This path is for a HTTP proxy, which is used for non-HTTPS requests.